opencti-platform/opencti-front/src/utils/defaultRepresentatives.ts

```diff
@@ -95,43 +95,49 @@ export const getMainRepresentative = (n: any, fallback = 'Unknown') => {
   if (typeof n.definition === 'object') {
     return defaultValueMarking(n);
   }
+
+  // Check for n.name explicitly and give it precedence.
+  if (n.name !== undefined) {
```

Comment:
Still don't get it. Name is the second choice of mainValue, so if is defined it will be applied...

Reply:
Because 'OR' operator returns the first truthy value it encounters.
 n.representative?.main = threat-actor--e984d89c-48e8-5c5e-8201-95517aff63bd, so n.name is not assigned.